apps/api/middleware/verifyToken.ts

```diff
@@ -0,0 +1,34 @@
+import admin from 'firebase-admin';
+
+export const environments = {
+	firebaseProjectId: process.env.FIREBASE_PROJECT_ID,
+	firebaseClientEmail: process.env.FIREBASE_CLIENT_EMAIL,
+	firebasePrivateKey: process.env.FIREBASE_PRIVATE_KEY,
+};
+
+admin.initializeApp({
+	credential: admin.credential.cert({
+		projectId: environments.firebaseProjectId,
+		clientEmail: environments.firebaseClientEmail,
+		privateKey: environments.firebasePrivateKey,
+	}),
+});
+
+export const verifyToken = async (req, res, next) => {
+	const authHeader = req.headers.authorization;
+	if (!authHeader) {
+		return res.status(401).send('Unauthorized');
+	}
+	const tokenParts = authHeader.split(' ');
+	if (tokenParts.length !== 2 || tokenParts[0] !== 'Bearer') {
+		return res.status(401).send('Unauthorized');
+	}
+	const idToken = tokenParts[1];
+	try {
+		const decodedToken = await admin.auth().verifyIdToken(idToken);
+		req.decodedToken = decodedToken;
+		next();
+	} catch (error) {
+		res.status(401).send('Unauthorized');
+	}
+};
```

Comment:
shouldn't handle `Unauthorized` here, would be something for the inner structure e.g GraphQL to handle; this middleware, however would inject "user" information to the request and leave it empty if there is any problem or no-authorization headers.